Content.Client/MassMedia/Ui/NewsReadMenu.xaml.cs

```diff
@@ -0,0 +1,54 @@
+using Content.Client.Message;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
+using Content.Shared.MassMedia.Systems;
+
+namespace Content.Client.MassMedia.Ui;
+
+[GenerateTypedNameReferences]
+public sealed partial class NewsReadMenu : DefaultWindow
+{
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    public event Action? NextButtonPressed;
+    public event Action? PastButtonPressed;
+
+    public NewsReadMenu(string name)
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+
+        if (Window != null)
+            Window.Title = name;
+
+        Next.OnPressed += _ => NextButtonPressed?.Invoke();
+        Past.OnPressed += _ => PastButtonPressed?.Invoke();
+    }
+
+    public void UpdateUI(NewsArticle article, int targetNum, int totalNum)
+    {
+        PageNum.Visible = true;
+        PageText.Visible = true;
+        ShareTime.Visible = true;
+
+        PageName.Text = article.Name;
+        PageText.SetMarkup(article.Content);
```

Comment:
Do you still need these markup sets on every update or can you just code them into the XML?